### PR TITLE
Update submariner to 0.14.0-m1

### PR DESCRIPTION
**Pull request by Jaanki**
Signed-off-by: Janki Chhatbar <jchhatba@redhat.com>

### CODE REVIEW

File: pkg/cloud/azure/azure.go [hunk 1]
```diff
@@ -88,11 +88,11 @@ func NewAzureProvider(
 
 	cloudInfo := azure.CloudInfo{
 		SubscriptionID: subscriptionID,
```
Comment (skitt) [inline pkg/cloud/azure/azure.go:90]: This needs an extra space:

```suggestion
		SubscriptionID:  subscriptionID,
```

File: Makefile [hunk 2]
```diff
@@ -111,7 +111,7 @@ ifeq (, $(shell which controller-gen))
 	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
 	cd $$CONTROLLER_GEN_TMP_DIR ;\
 	go mod init tmp ;\
-	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.5.0 ;\
+	go get sigs.k8s.io/controller-tools/cmd/controller-gen@v0.12.3 ;\
```
Comment (tpantelis) [inline Makefile:114]: Why was this version bumped, how is it related to the submariner-operator bump to 0.14.0-m1?

File: deploy/olm-catalog/manifests/submariner-addon.clusterserviceversion.yaml [hunk 3]
```diff
@@ -1,326 +0,0 @@
-apiVersion: operators.coreos.com/v1alpha1
```
Comment (tpantelis) [inline deploy/olm-catalog/manifests/submariner-addon.clusterserviceversion.yaml:1]: Why was this file and the other two deleted?

Comment (Jaanki) [reply deploy/olm-catalog/manifests/submariner-addon.clusterserviceversion.yaml:1]: I didn't delete anything explicitly. I just bumped the version and ran `go mod tidy`. `go mod vendor`. All the other changes are automatic.

Comment (tpantelis) [reply deploy/olm-catalog/manifests/submariner-addon.clusterserviceversion.yaml:1]: But the commit is intended to bump the **submariner-operator** version so how/why did that result in changes to the add-on manifest/deployment? Specifically the `controller-gen.kubebuilder.io/version` was bumped in the yaml files - did you specifically change that? If not then what specific change in the _go.mod_  did it and is it desired?

**Comment by tpantelis** [comment]
Superseded by https://github.com/stolostron/submariner-addon/pull/534.

**Pull request closed without merging**